src/new_json.py

```diff
@@ -0,0 +1,247 @@
+            get_replace_columns(table['dialect'])
+            get_nulls(table['dialect'])
+            get_delimiter(table['dialect'])
+            get_do_not_bulk_insert(table['dialect'])
+            get_contains_pk(table['dialect'])
+            get_fixed_width(table['dialect'])
+            get_header_rows(table['dialect'])
+
+            # set table schema
+            table['schema'] = {}
+            table['schema']["fields"] = []
+            print("Enter columns [format = name, type, (optional) size] (press return to skip):\n\n")
+            while True:
+                # get column list (optional)
+                try:
+                    col_list = clean_input("", split_char=',', ignore_empty=True)
+                    if not col_list:
+                        break
+                    if not isinstance(col_list, list):
+                        raise Exception
+
+                    col_list = [c.strip() for c in col_list]
+                    col_obj = dict()  # dict to store column data
+                    col_obj["name"] = col_list[0]
+                    col_obj["type"] = col_list[1]
+
+                    if len(col_list) > 2:
+                        if type(eval(col_list[2])) != int:
+                            raise Exception
+                        col_obj["size"] = col_list[2]
+                    table["schema"]["fields"].append(col_obj)
+                except:
+                    print("Exception occured. Check the input format again.\n")
+                    pass
+
+            isCT = clean_input(
+                "Add crosstab columns? (y,N): ", ignore_empty=True)
+            if isCT.lower() in ["y", "yes"]:
+                ct_column = clean_input("Crosstab column name: ")
+                ct_names = []
+                print("Enter names of crosstab column values (Press return after each name):\n")
+                name = clean_input()
+                while name != "":
+                    ct_names.append(name)
+                    name = clean_input()
+
+                table['schema']['ct_column'] = ct_column
+                table['schema']['ct_names'] = ct_names
+
+            contents['resources'].append(table)
+    give_message = clean_input(
+        "Would you like to add a Message? (y,N): ", ignore_empty=True)
+    if give_message.lower() in ["y", "yes"]:
+        contents['message'] = clean_input("Provide your Message: ", ignore_empty=True)
+    contents['urls'] = tableurls
+    file_name = contents['name'] + ".json"
+    file_name = file_name.replace('-', '_')
+    with open(os.path.join('scripts', file_name), 'w') as output_file:
+        json.dump(contents,
+                  output_file,
+                  sort_keys=True,
+                  indent=4,
+                  separators=(',', ': '))
+        output_file.write('\n')
+        print("\nScript written to " + file_name)
+        output_file.close()
+
+
+if __name__ == '__main__':
+    create_json()
```

Comment:
This entire file has also been shipped from retriever's [`datapackage.py`](https://github.com/weecology/retriever/blob/master/retriever/lib/datapackage.py).